packages/suite/src/reducers/suite/suiteReducer.ts

```diff
@@ -443,12 +440,10 @@ export const selectHasExperimentalFeature =
     (feature: ExperimentalFeature) => (state: SuiteRootState) =>
         state.suite.settings.experimental?.includes(feature) ?? false;
 
-type StateForFirmwareChecks = SuiteRootState & DeviceRootState & MessageSystemRootState;
-
 /**
  * Get firmware revision check error, or null if check was successful / skipped.
  */
-export const selectFirmwareRevisionCheckError = (state: StateForFirmwareChecks) => {
+export const selectFirmwareRevisionCheckError = (state: AppState) => {
```

Comment:
`AppState` describes more accurately what is expected than `StateForFirmwareChecks`